docs/docsite/rst/guide_vardict.rst

```diff
@@ -0,0 +1,176 @@
+..
+  Copyright (c) Ansible Project
+  GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+  SPDX-License-Identifier: GPL-3.0-or-later
+
+.. _ansible_collections.community.general.docsite.guide_vardict:
+
+VarDict Guide
+=============
+
+Introduction
+^^^^^^^^^^^^
+
+The ``ansible_collections.community.general.plugins.module_utils.vardict`` module util provides the
+``VarDict`` class to help manage the module variables. That class is a container for module variables,
+especially the ones for which the module must keep track of state changes, and the ones that should
+be published as return values.
+
+Each variable has extra behaviors controlled by associated metadata, simplifying the generation of
+output values from the module.
+
+Quickstart
+""""""""""
+
+The simplest way of using ``Vardict`` is:
+
+.. code-block:: python
+
+    from ansible_collections.community.general.plugins.module_utils.vardict import VarDict
+
+Then in ``main()``, or any other function called from there:
+
+.. code-block:: python
+
+    vars = VarDict()
+
+    # Next 3 statements are equivalent
+    vars.abc = 123
+    vars["abc"] = 123
+    vars.set("abc", 123)
+
+    vars.xyz = "bananas"
+    vars.ghi = False
+
+And by the time the module is about to exit:
+
+.. code-block:: python
+
+    results = vars.output()
+    module.exit_json(**results)
+
+That will make the return value of the module:
+
+.. code-block:: javascript
+
+    {
+        "abc": 123,
+        "xyz": "bananas",
+        "ghi": false
+    }
+
+Metadata
+""""""""
+
+The metadata values associated with each variable are:
+
+- ``output`` - marks the variable for module output as a module return value.
+- ``fact`` - marks the variable for module output as an Ansible fact.
+- ``verbosity`` - sets the minimum level of verbosity for which the variable will be included in the output.
+- ``change`` - controls the detection of changes in the variable value.
+- ``initial_value`` - when using ``change`` and need to forcefully set an intial value to the variable.
+- ``diff`` - used along with ``change``, this will generate an Ansible-style diff ``dict``.
```

Comment:
Maybe specify the types for these metadata as well. For `verbosity` it's clear it's an `int` and some others are obviously `bool`s, but what's `initial_value` for example? Is it a boolean? An arbitrary value?

Reply:
Is there a recommended way to do that? Maybe:
```suggestion
- ``diff: bool`` - used along with ``change``, this will generate an Ansible-style diff ``dict``.
```
??

Reply:
Not that I know of, but `diff: bool` or `diff (bool)` or something like that is probably fine.